Improve helm chart: Allow to enable or disable cloudControllerManager/cloudNodeManager, configure ca certificates directory and replicas

forselli-stratio opened the pull request:
…/cloudNodeManager, configure ca certificates directory and replicas

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

/kind feature

#### What this PR does / why we need it:
This PR allows to enable/disable both cloud-controller-manager or cloud-node-manager deployments. Also, you can configure a different directory for the CA certificates in cloud-controller-manager. Finally, it allows you to configure the number of replicas for the cloud-controller-manager.



#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

jackfrancis commented on helm/cloud-provider-azure/Chart.yaml:
```diff
@@ -13,4 +13,4 @@ maintainers:
 sources:
 - https://github.com/kubernetes-sigs/cloud-provider-azure
-version: 1.23.11
+version: 1.24.0
```
Let's skip the chart version update for now, we can queue up these changes and release them with the next batch of cloud-provider-azure releases (which will include 1.23.12)

jackfrancis commented on helm/cloud-provider-azure/values.yaml:
```diff
@@ -30,6 +32,7 @@ cloudControllerManager:
   # minResyncPeriod: "100m"
   # nodeStatusUpdateFrequency: "10m"
   port: 10267
+  replicaCount: 1
```
slight preference to just using the value `replicas` here (to match the standard k8s spec)

forselli-stratio replied:
sure, no problem

jackfrancis commented:
Thanks for this @forselli-stratio! Added some comments, overall looking good.

forselli-stratio commented:
> Thanks for this @forselli-stratio! Added some comments, overall looking good.

Hi @jackfrancis , i updated the PR with the changes you sugested. Thank you!

MartinForReal commented:
/ok-to-test

lzhecheng commented:
@forselli-stratio please fix the `pull-cloud-provider-azure-check` error. Thanks.

lzhecheng commented on helm/cloud-provider-azure/README.md:
```diff
@@ -70,6 +72,7 @@ Below is the complete set of configuration that you may include when invoking `h
 | `cloudControllerManager.imagePullPolicy` | `"IfNotPresent"` | you may change to`"Always"` or `"Never"` if appropriate for your environment, see [here](https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy) for more info |
 | `cloudControllerManager.logVerbosity` | `"2"` | set to a higher number when debugging the azure-cloud-controller-manager runtime |
 | `cloudControllerManager.port` | `"10267"` | TCP port on which azure-cloud-controller-manager pod responds to requests |
+| `cloudControllerManager.replicaCount` | `1` | Number of replicas for the azure-cloud-controller-manager deployment. |
```
```suggestion
| `cloudControllerManager.replicaCount` | `1` | Number of replicas for the azure-cloud-controller-manager deployment. It should be no more than the number of control plane Nodes |
```
I suggest mentioning this.

lzhecheng replied:
Also, isn't it `cloudControllerManager.replicas`?

forselli-stratio replied:
Fixed

lzhecheng commented:
> @forselli-stratio please fix the `pull-cloud-provider-azure-check` error. Thanks.

The failure results from verify helm script. Not this PR.

MartinForReal commented:
/override pull-cloud-provider-azure-check

jackfrancis commented:
@MartinForReal @lzhecheng @forselli-stratio 

FYI: the failure of the `pull-cloud-provider-azure-check` PR is in fact due to this PR. As documented here, every time you make an update to the helm chart, you must run the `hack/update-helm-repo.sh` script to publish the new chart release to the repo:

[hack/update-helm-repo.sh](https://github.com/kubernetes-sigs/cloud-provider-azure/tree/master/helm/cloud-provider-azure#maintaining-the-repo)

I also realize that my prior guidance around not cutting a new version with these changes is wrong, apologies for that.

I have a new PR that attempts to clarify documentation around helm chart maintenance here:

https://github.com/kubernetes-sigs/cloud-provider-azure/pull/1658

jackfrancis commented:
So, @forselli-stratio I was wrong about not cutting a new version. Can we update the version in `Chart.yaml` to "1.23.11+20220505" (assuming you make these changes today, May 5).

After updating the version, please run `make update-helm` to render the new chart updates as a release in the local repo.

Thanks!

nilo19 commented on helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:
```diff
@@ -277,10 +278,12 @@ spec:
             path: /etc/kubernetes
         - name: etc-ssl
```
would you mind changing this name as well?

jackfrancis replied:
Now that the path is configurable we could change the name key to "ssl-mount" (remember to change both instances)

Thanks!

forselli-stratio replied:
Done

forselli-stratio commented:
Hi @jackfrancis @MartinForReal @lzhecheng , i applied your suggestions. I noticed that 3 days ago you changed the version to 1.24.0, so i'm not sure if the version i used is correct. Thank you!!

jackfrancis commented on helm/cloud-provider-azure/Chart.yaml:
```diff
@@ -13,4 +13,4 @@ maintainers:
 sources:
 - https://github.com/kubernetes-sigs/cloud-provider-azure
-version: 1.24.0
+version: 1.24.1+20220509
```
This should be 1.24.**0**+<datestamp> because the current, latest release of the chart is 1.24.0.

forselli-stratio replied:
Done @jackfrancis , thanks!

jackfrancis commented:
@forselli-stratio the versions are still not right. To correct:

1)

```
$ # remove the 1.24.1 chart from the repo
$ rm helm/repo/cloud-provider-azure-1.24.1+20220509.tgz
```

2)

```
$ # update the Chart.yaml to 1.24.0+<current date>
$ cat helm/cloud-provider-azure/Chart.yaml | grep 'version:'
version: 1.24.0+20220519
```

3)

```
$ # update the repo with the new 1.24.0 version
$ hack/update-helm-repo.sh 
Successfully packaged chart and saved it to: helm/repo/new/cloud-provider-azure-1.24.0+20220519.tgz
```

4)

```
$ # manually remove the 1.24.1 entry from index.yaml, something like this:
$ git diff helm/repo/index.yaml 
diff --git a/helm/repo/index.yaml b/helm/repo/index.yaml
index 351ee5134..058cc1b31 100644
--- a/helm/repo/index.yaml
+++ b/helm/repo/index.yaml
@@ -2,10 +2,10 @@ apiVersion: v1
 entries:
   cloud-provider-azure:
   - apiVersion: v1
-    created: "2022-05-09T13:30:32.197485547+02:00"
+    created: "2022-05-19T09:31:26.411883-07:00"
     description: A Helm chart for installing kubernetes-sigs/cloud-provider-azure
       components
-    digest: 081cc30534a47c6d0d420462cc81375453b56d1ca1ae1d3f17a7e7d6cc7f6508
+    digest: 72a94ef39f105f869fb9f40163fc96a19d8edcb1788746b3dae3dd3687fcaf8d
     home: https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/cloud-provider-azure/README.md
     keywords:
     - cloud-provider-azure
@@ -20,29 +20,8 @@ entries:
     sources:
     - https://github.com/kubernetes-sigs/cloud-provider-azure
     urls:
-    - cloud-provider-azure-1.24.1+20220509.tgz
-    version: 1.24.1+20220509
-  - apiVersion: v1
-    created: "2022-05-06T10:10:12.501287-07:00"
-    description: A Helm chart for installing kubernetes-sigs/cloud-provider-azure
-      components
-    digest: c9aa0bf18ee1442f9ae57ae277a6a7b1c7b88f33836508a6a97fb593cd1e1750
-    home: https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/cloud-provider-azure/README.md
-    keywords:
-    - cloud-provider-azure
-    - kubernetes-sigs
-    - kubernetes
-    maintainers:
-    - email: jackfrancis@gmail.com
-      name: Jack Francis
-    - email: zhechengli1995@outlook.com
-      name: Zhecheng Li
-    name: cloud-provider-azure
-    sources:
-    - https://github.com/kubernetes-sigs/cloud-provider-azure
-    urls:
-    - cloud-provider-azure-1.24.0.tgz
-    version: 1.24.0
+    - cloud-provider-azure-1.24.0+20220519.tgz
+    version: 1.24.0+20220519
   - apiVersion: v1
     created: "2022-05-02T14:33:56.285027-07:00"
     description: A Helm chart for installing kubernetes-sigs/cloud-provider-azure
@@ -106,4 +85,4 @@ entries:
     urls:
     - cloud-provider-azure-1.23.8.tgz
     version: 1.23.8
-generated: "2022-05-09T13:30:32.196495952+02:00"
+generated: "2022-05-19T09:31:26.411046-07:00"
```

forselli-stratio commented:
Hi @jackfrancis , thanks for the detailed explanation, i hope i understood the flow right this time.

lzhecheng commented on helm/cloud-provider-azure/README.md:
```diff
@@ -70,6 +72,7 @@ Below is the complete set of configuration that you may include when invoking `h
 | `cloudControllerManager.imagePullPolicy` | `"IfNotPresent"` | you may change to`"Always"` or `"Never"` if appropriate for your environment, see [here](https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy) for more info |
 | `cloudControllerManager.logVerbosity` | `"2"` | set to a higher number when debugging the azure-cloud-controller-manager runtime |
 | `cloudControllerManager.port` | `"10267"` | TCP port on which azure-cloud-controller-manager pod responds to requests |
+| `cloudControllerManager.replicas` | `1` | Number of replicas for the azure-cloud-controller-manager deployment. It should be no more than the number of control plane Nodes |
```
```suggestion
| `cloudControllerManager.replicas` | `"1"` | Number of replicas for the azure-cloud-controller-manager deployment. It should be no more than the number of control plane Nodes |
```

lzhecheng reviewed (commented):
@forselli-stratio Thank you for the PR! Please rebase again and I think this PR should be merged then.

forselli-stratio commented:
Hi @lzhecheng , done!!

lzhecheng commented:
/lgtm
/approve
